cocotb_build_libs.py

```diff
@@ -6,17 +6,55 @@
 import sys
 import sysconfig
 import logging
-import distutils
+import re
+import shutil
 import subprocess
 import textwrap
 
 from setuptools import Extension
-from distutils.spawn import find_executable
 from setuptools.command.build_ext import build_ext as _build_ext
-from distutils.file_util import copy_file
-from distutils.ccompiler import get_default_compiler
 from typing import List
 
+# vendored in from distutils/ccompiler.py:
```

Comment:
Are you sure this is necessary? I was under the impression that `setuptools` has a copy of `distutils` which is what gets imported when setup.py is run.

Reply:
See [this](https://github.com/pypa/setuptools/blob/0c302d232dd1a1f4ec531050532a84274edac2f4/_distutils_hack/__init__.py#L49) import hack in setuptools. I don't think we need to worry about vendoring in distutils code that is used in setup.py. We must only ensure that setuptools is imported first.

Reply:
Ah yes, this works. Let's see if CI confirms...